### PR TITLE
[Doc] Improve LightningTrainer advanced example

### PR DESCRIPTION
**Pull request by woshiyyya**
## Why are these changes needed?

Addressed the comments left from the previous PR [#34082 ].

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/train/examples/lightning/lightning_cola_advanced.ipynb [hunk 1]
```diff
@@ -40,11 +40,43 @@
...
+  },
   {
    "cell_type": "code",
-   "execution_count": 3,
+   "execution_count": 6,
    "metadata": {},
    "outputs": [],
+   "source": [
+    "# !pip install numpy datasets \"transformers>=4.19.1\" \"pytorch_lightning>=1.6.5\""
+   ]
```
Comment (matthewdeng) [inline doc/source/train/examples/lightning/lightning_cola_advanced.ipynb:57]: Any particular reason that this needs to be commented out?

File: doc/source/train/examples/lightning/lightning_cola_advanced.ipynb [hunk 2]
```diff
@@ -40,11 +40,43 @@
...
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/ray/anaconda3/lib/python3.9/site-packages/xgboost/compat.py:31: FutureWarning: pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.\n",
+      "  from pandas import MultiIndex, Int64Index\n"
+     ]
+    }
+   ],
    "source": [
```
Comment (matthewdeng) [inline doc/source/train/examples/lightning/lightning_cola_advanced.ipynb:80]: I think we can exclude these types of outputs that aren't valuable for the user.

File: doc/source/train/examples/lightning/lightning_cola_advanced.ipynb [hunk 3]
```diff
@@ -272,13 +515,15 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
+    "## 4. Fine-tune the model with LightningTrainer\n",
```
Comment (matthewdeng) [inline doc/source/train/examples/lightning/lightning_cola_advanced.ipynb:518]: nit: For consistency with other docs we don't need to have numbers in the headers (`4.`) Same applies for the other steps.

**Review by matthewdeng — APPROVED** [review]

File: doc/source/train/examples/lightning/lightning_cola_advanced.ipynb (hunk 1 above)
Comment (woshiyyya) [reply doc/source/train/examples/lightning/lightning_cola_advanced.ipynb]: nope, just followed what other examples did. 😂 I'll uncomment this

File: doc/source/train/examples/lightning/lightning_cola_advanced.ipynb (hunk 3 above)
Comment (woshiyyya) [reply doc/source/train/examples/lightning/lightning_cola_advanced.ipynb]: Ok! Fixed

File: doc/source/train/examples/lightning/lightning_cola_advanced.ipynb (hunk 2 above)
Comment (woshiyyya) [reply doc/source/train/examples/lightning/lightning_cola_advanced.ipynb]: Done

File: doc/source/train/examples/lightning/lightning_cola_advanced.ipynb [hunk 4]
```diff
@@ -40,9 +40,32 @@
     "SMOKE_TEST = True"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Uncomment and run the following line in order to install all the necessary dependencies:"
```
Comment (matthewdeng) [inline doc/source/train/examples/lightning/lightning_cola_advanced.ipynb:47]:
```suggestion
    "Run the following line in order to install all the necessary dependencies:"
```

File: doc/source/train/examples/lightning/lightning_cola_advanced.ipynb [hunk 5]
```diff
@@ -291,147 +527,1163 @@
...
+   "source": [
+    "result"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 5. Do Batch Inference with a Saved Checkpoint"
```
Comment (matthewdeng) [inline doc/source/train/examples/lightning/lightning_cola_advanced.ipynb:1580]:
```suggestion
    "## Do Batch Inference with a Saved Checkpoint"
```

File: doc/source/train/examples/lightning/lightning_cola_advanced.ipynb [hunk 6]
```diff
@@ -87,9 +228,21 @@
...
-   "execution_count": null,
+   "execution_count": 9,
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "find: ‘.git’: No such file or directory\n",
```
Comment (matthewdeng) [inline doc/source/train/examples/lightning/lightning_cola_advanced.ipynb:238]: Remove this block

**Pull request merged**